### ISSUE TITLE
Siamese Tutorial last example introduce a validation leak.

### ISSUE THREAD

**Issue by kessido** [fastai/fastai#2889]
In the Siamese tutorial the last part introduce an example of using the datablock infrastructure along with a custom ImageTupleBlock.

```
siamese = DataBlock(
    blocks=(ImageTupleBlock, CategoryBlock),
    get_items=get_tuple_files,
    get_x=get_x, get_y=get_y,
    splitter=RandomSplitter(),
    item_tfms=Resize(224),
    batch_tfms=[Normalize.from_stats(*imagenet_stats)]
)
```

In this example we are using `get_tuple_files` to build this dataset.
```
def get_tuple_files(path):
    files = get_image_files(path)
    return [[f, *draw_other(f)] for f in files]
```

This uses all files for the split, and does not separate the valid \ train files, only different pairs are used. 
The image in the validation set are examples which are now presented in the training set as well.

**Comment by jph00** [comment]
That's true. I'd be happy to take a PR if you're interested in fixing this in a similar way to the earlier example.
